### PR TITLE
[FLINK-25546][flink-connector-base][JUnit5 Migration] Module: flink-connector-base

### PR DESCRIPTION
**Pull request by snuyanzin**
## What is the purpose of the change

Migrate flink-connector-base module to AssertJ and JUnit 5 following the [JUnit 5 Migration Guide](https://docs.google.com/document/d/1514Wa_aNB9bJUen4xm5uiuXOooOJTtXqS_Jqk9KJitU/edit)

## Brief change log
Use JUnit5 and AssertJ in tests instead of JUnit4 and Hamcrest

## Verifying this change
This change is a code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: ( no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper:  no)
  - The S3 file system connector: ( no)

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not applicable )

### CODE REVIEW

**Comment by snuyanzin** [comment]
@flinkbot run azure

**Comment by snuyanzin** [comment]
@Jiabao-Sun since you are one of those who contributed a lot in JUnit5 migration may I ask you to have a look here?

File: flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/AsyncSinkBaseITCase.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 /** Integration tests of a baseline generic sink that implements the AsyncSinkBase. */
 @ExtendWith(TestLoggerExtension.class)
```
Comment (Jiabao-Sun) [inline flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/AsyncSinkBaseITCase.java:34]:
```suggestion
```

**Review by Jiabao-Sun — COMMENTED** [review]
Thanks @snuyanzin for this great work.
I just left one comment.

File: flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/AsyncSinkBaseITCase.java (hunk 1 above)
Comment (snuyanzin) [reply flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/AsyncSinkBaseITCase.java]: make sense, 
also cleaned imports

thanks!

**Pull request merged**